### PR TITLE
BREAKING(fmt/duration): rename prettyDuration to format

### PR DESCRIPTION
**Pull request by timreichen**
<del>- adds `parse` function to reverse formatting</del>

The following name changes are inspired by https://github.com/tc39/proposal-intl-duration-format 
- renames `prettyDuration()` to `format()`
- renames `formatType` to `style`
- renames `time`  formatType to `digital`
- renames `short`  formatType to `narrow`

### CODE REVIEW

File: fmt/README.md [hunk 1]
```diff
@@ -270,22 +270,40 @@ prettyBytes(1337, { locale: "de" });
...
 // "1m 39s 674ms"
-prettyDuration(99674, { ignoreZero: true });
+format(99674, { ignoreZero: true });
 
 // "1 minutes, 39 seconds, 674 milliseconds"
-prettyDuration(99674, { formatType: "full", ignoreZero: true });
+format(99674, { style: "full", ignoreZero: true });
+```
+
+parse time duration to milliseconds.
```
Comment (cjihrig) [inline fmt/README.md:291]:
```suggestion
Parse time duration to milliseconds.
```

File: fmt/README.md [hunk 2]
```diff
@@ -270,22 +270,40 @@ prettyBytes(1337, { locale: "de" });
 
 # Pretty Duration
 
-Get formatted time duration from milliseconds.
+format milliseconds to time duration.
```
Comment (cjihrig) [inline fmt/README.md:273]:
```suggestion
Format milliseconds to time duration.
```

File: fmt/duration.ts [hunk 3]
```diff
@@ -25,7 +25,7 @@ const keyList: Record<keyof DurationObj, string> = {
 };
 
 /** Parse milleseconds into a duration. */
-function parseDuration(ms: number): DurationObj {
+function millisecondsToObject(ms: number): DurationObject {
```
Comment (cjihrig) [inline fmt/duration.ts:28]:
```suggestion
function millisecondsToDurationObject(ms: number): DurationObject {
```

File: fmt/duration.ts [hunk 4]
```diff
@@ -53,33 +53,45 @@ function durationArray(
   ];
 }
 
+function objectToMilliseconds(object: DurationObject) {
```
Comment (cjihrig) [inline fmt/duration.ts:56]:
```suggestion
function durationObjectToMilliseconds(object: DurationObject) {
```

File: fmt/duration.ts [hunk 5]
```diff
@@ -121,7 +133,75 @@ export function prettyDuration(
...
+    ms: 0,
+    us: 0,
+    ns: 0,
+  };
+  let match;
+  const valueRegex = /(?<value>\d+)\s*(?<name>[\µa-z]+)/g;
+  while ((match = valueRegex.exec(value)) !== null) {
+    const groups = match.groups!;
+    const name = groups.name;
+    const value = parseInt(groups.value);
```
Comment (cjihrig) [inline fmt/duration.ts:167]:
```suggestion
    const value = Number.parseInt(groups.value, 10);
```

File: fmt/duration_test.ts [hunk 6]
```diff
@@ -1,64 +1,113 @@
...
 
 Deno.test({
-  name: "short duration ignore zero",
+  name: "format narrow duration ignore zero",
   fn() {
-    assertEquals(prettyDuration(99674, { ignoreZero: true }), "1m 39s 674ms");
+    assertEquals(format(99674, { ignoreZero: true }), "1m 39s 674ms");
+  },
+});
+
```
Comment (cjihrig) [inline fmt/duration_test.ts:65]: Can you add some tests to verify that `parse(format(value)) === value`.

Comment (timreichen) [reply fmt/duration_test.ts:65]: done

**Comment by kt3k** [comment]
I'm not in favor of adding this feature to std. What this `parse` parses are random duration format, and doesn't look useful to me.

**Comment by retraigo** [comment]
I didn't want to add `parse` in my earlier PR because I didn't feel like it needed to be part of a "standard library", considering people represent durations in a lot of ways.

There is a more complete version of `parse` in [duration.js][(https://github.com/retraigo/duration.js](https://github.com/retraigo/duration.js/blob/main/src/duration.ts#L560)) (though the module itself is quite big as a result) if you would like to refer.

Apart from renaming `prettyDuration` to `format` (had it as `prettyDuration` to maintain consistency with `prettyBytes`, the other renamings seem helpful.

**Comment by timreichen** [comment]
> I'm not in favor of adding this feature to std. What this `parse` parses are random duration format, and doesn't look useful to me.

I am not quite following you, what do you mean exactly by random? It parses the possible outputted formats of `prettyDuration`, or did I miss something?
I think it is useful to provide a solution to reverse a formatting process, similar to what we do in std/datetime. There is no harm in providing it.

> I didn't want to add `parse` in my earlier PR because I didn't feel like it needed to be part of a "standard library", considering people represent durations in a lot of ways.
> 
> There is a more complete version of `parse` in [duration.js](https://github.com/retraigo/duration.js/blob/main/src/duration.ts#L560) (though the module itself is quite big as a result) if you would like to refer.

That is true, but `parse` supports the current formatting options.  Though it could expanded, I don't think it needs to provide more than what the counterpart function is capable of formatting atm.

> Apart from renaming `prettyDuration` to `format` (had it as `prettyDuration` to maintain consistency with `prettyBytes`, the other renamings seem helpful.

I think it would work better the other way around. Web APIs have some consistent naming schemes, `format` is one of them and I think we should adapt them, so `prettyBytes` would be better of called `format` as well imo.

**Comment by kt3k** [comment]
> > I'm not in favor of adding this feature to std. What this `parse` parses are random duration format, and doesn't look useful to me.
> 
> I am not quite following you, what do you mean exactly by random? It parses the possible outputted formats of `prettyDuration`, or did I miss something? I think it is useful to provide a solution to reverse a formatting process, similar to what we do in std/datetime. There is no harm in providing it.

It's random because it doesn't have any formal specification. What is correct format or what is bug is not well-defined here. Users might report bugs about unspecified/untested format and we need to address those. That'll cost maintenance effort.

**Comment by timreichen** [comment]
@kt3k Ok, I removed the parse function for now.

**Review by cjihrig — APPROVED** [review]

**Comment by retraigo** [comment]
I think `prettyBytes` should be updated to match this naming convention too.

**Comment by timreichen** [comment]
> I think `prettyBytes` should be updated to match this naming convention too. Maybe in another PR...

I'll open a separate PR 👍

**Comment by cjihrig** [comment]
@kt3k are you still opposed to this after the latest changes?

**Review by kt3k — APPROVED** [review]
I'm now in favor. `format` sounds more 'standard' than `prettyDuration`. I think this change is worth a breaking change.

LGTM

**Pull request merged**